src/util.ts

```diff
@@ -220,9 +221,10 @@ export function merge(target: object, obj: object, options?: MergeOptions): obje
  *
  * @param  {Array} xs The first array
  * @param  {Array} ys The second array
+ * @param  {Function} customHashFunction Custom function to hash values, default the hashCode method
  * @return {Array}    Result array
  */
-export function intersection(a: any[], b: any[]): any[] {
+export function intersection(a: any[], b: any[], customHashFunction: CustomHashFunction): any[] {
```

Comment:
`hashFunction` should be optional. same for the other affected functions; `union`, `memoize`, and `groupBy`.

Reply:
done